sdks/unity/model/GameServer.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 Google LLC
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Agones.Model
+{
+    /// <summary>
+    /// A GameServer Custom Resource Definition object We will only export those resources that make the most sense. Can always expand to more as needed.
+    /// </summary>
+    public class GameServer :  IEquatable<GameServer>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GameServer" /> class.
+        /// </summary>
+        public GameServer(IReadOnlyDictionary<string, object> data)
+        {
+            this.ObjectMeta = new GameServerObjectMeta((Dictionary<string,object>) data["object_meta"]);
+            this.Spec = new GameServerSpec((Dictionary<string,object>) data["spec"]);
```

Comment:
Why this requires force conversion (throw exception if not converted) while status requires soft conversion with `as` (null value if not converted)? If intended can you add a comment to document?